Sources/Sentry/SentryTracer.m

```diff
@@ -469,7 +483,18 @@ - (void)finishInternal
     }];
 
     @synchronized(_children) {
-        if (self.idleTimeout > 0.0 && _children.count == 0) {
+        if ([self hasIdleTimeout] && _children.count == 0) {
+            SENTRY_LOG_DEBUG(
+                @"Was waiting for timeout for UI event trace and no children to wait for.");
+            if ([self didAutomaticTraceExceedMaxDuration]) {
+                SENTRY_LOG_DEBUG(@"Transaction exceeded max duration.");
+                return;
+            }
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+            [SentryProfiler stopProfilingSpan:self.rootSpan];
+            [SentryProfiler linkTransaction:_transaction];
+#endif // SENTRY_TARGET_PROFILING_SUPPORTED
+            [_hub captureTransaction:_transaction withScope:_hub.scope];
```

Comment:
`h`: We are missing `[self addMeasurementsAndChildrenToTransaction:_transaction];`. That code here is very similar to the following lines https://github.com/getsentry/sentry-cocoa/blob/7d63c3dd33521e6fe1e7e695599f916a528a5ea3/Sources/Sentry/SentryTracer.m#L516-L533

Can we extract a function to avoid such mistakes?